sdk/include/psdk/chstring.h

```diff
@@ -6,125 +6,372 @@
+        AssignCopy(nSrcLen, reinterpret_cast<CHSTRING_LPCWSTR>(lpszSrcData));
+    }
+    void ConcatCopy(int nSrc1Len, CHSTRING_LPCWSTR lpszSrc1Data, int nSrc2Len, CHSTRING_LPCWSTR lpszSrc2Data);
+    void ConcatCopy(int nSrc1Len, const wchar_t* lpszSrc1Data, int nSrc2Len, const wchar_t* lpszSrc2Data)
+    {
+        ConcatCopy(nSrc1Len, reinterpret_cast<CHSTRING_LPCWSTR>(lpszSrc1Data), nSrc2Len, reinterpret_cast<CHSTRING_LPCWSTR>(lpszSrc2Data));
+    }
+    void ConcatInPlace(int nSrcLen, CHSTRING_LPCWSTR lpszSrcData);
+    void ConcatInPlace(int nSrcLen, const wchar_t* lpszSrcData)
+    {
+        ConcatInPlace(nSrcLen, reinterpret_cast<CHSTRING_LPCWSTR>(lpszSrcData));
+    }
     void CopyBeforeWrite();
     CHStringData* GetData() const;
     void Init();
-    int LoadStringW(UINT nID, LPWSTR lpszBuf, UINT nMaxBuf);
+    int LoadStringW(UINT nID, CHSTRING_LPWSTR lpszBuf, UINT nMaxBuf);
     void Release();
     static void WINAPI Release(CHStringData* pData);
-    static int WINAPI SafeStrlen(LPCWSTR lpsz);
+    static int WINAPI SafeStrlen(CHSTRING_LPCWSTR lpsz);
+    static int WINAPI SafeStrlen(const wchar_t* lpsz)
+    {
+        return SafeStrlen(reinterpret_cast<CHSTRING_LPCWSTR>(lpsz));
+    }
 };
 
-inline BOOL operator==(const CHString& s1, LPCWSTR s2) { return s1.Compare(s2) == 0; }
+inline BOOL operator==(const CHString& s1, CHSTRING_LPCWSTR s2) { return s1.Compare(s2) == 0; }
 inline BOOL operator==(const CHString& s1, const CHString& s2) { return s1.Compare(s2) == 0; }
 
-inline BOOL operator!=(const CHString& s1, LPCWSTR s2) { return s1.Compare(s2) != 0; }
+inline BOOL operator!=(const CHString& s1, CHSTRING_LPCWSTR s2) { return s1.Compare(s2) != 0; }
 inline BOOL operator!=(const CHString& s1, const CHString& s2) { return s1.Compare(s2) != 0; }
 
-inline BOOL operator<(const CHString& s1, LPCWSTR s2) { return s1.Compare(s2) < 0; }
+inline BOOL operator<(const CHString& s1, CHSTRING_LPCWSTR s2) { return s1.Compare(s2) < 0; }
 inline BOOL operator<(const CHString& s1, const CHString& s2) { return s1.Compare(s2) < 0; }
 
-inline BOOL operator>(const CHString& s1, LPCWSTR s2) { return s1.Compare(s2) > 0; }
+inline BOOL operator>(const CHString& s1, CHSTRING_LPCWSTR s2) { return s1.Compare(s2) > 0; }
 inline BOOL operator>(const CHString& s1, const CHString& s2) { return s1.Compare(s2) > 0; }
 
-inline BOOL operator<=(const CHString& s1, LPCWSTR s2) { return s1.Compare(s2) <= 0; }
+inline BOOL operator<=(const CHString& s1, CHSTRING_LPCWSTR s2) { return s1.Compare(s2) <= 0; }
 inline BOOL operator<=(const CHString& s1, const CHString& s2) { return s1.Compare(s2) <= 0; }
 
-inline BOOL operator>=(const CHString& s1, LPCWSTR s2) { return s1.Compare(s2) >= 0; }
+inline BOOL operator>=(const CHString& s1, CHSTRING_LPCWSTR s2) { return s1.Compare(s2) >= 0; }
 inline BOOL operator>=(const CHString& s1, const CHString& s2) { return s1.Compare(s2) >= 0; }
 
+/* Have GCC link to the symbols exported by framedyn.dll */
+#ifdef __GNUC__
```

Comment:
Why can't we hide this in the source code of framedyn itself as it previously was?

Reply:
Because you might want to link your code to an import lib generated outside of the ReactOS tree.